### DISCUSSION TITLE
Please support WIP Limits across multiple columns

### DISCUSSION THREAD

**Discussion by erinmez** [community/community#103010] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects

### Body

It was discussed here and I think it is time to revisit the topic:
https://github.com/orgs/community/discussions/4848

In order to apply "Kanban", it's necessary for me to have work signaling properly.  In "Kanban" an On-Going and Done column for each area/step (e.g. Development) is used to signal and buffer work correctly by limiting work across BOTH columns. 

The problems with the current approach to have WIP-Limits on an "On-Going" or "Done" is as follows:

- if the WIP limit is only on the on-going column:
people can fill up the done column & there is no incentive to empty the done column by moving items to the next step's on-going column

- if the WIP limit is only on the done column:
people can start new work items as they please with no boundary and once they are finished they are breaching the WIP limit of done.
Thus the signaling is wrong, they should not have started the work in the first place.

The WIP Limit should be rather across those columns:
<img width="609" alt="wiplimits" src="https://github.com/community/community/assets/16493423/fb81ed6a-fe4d-49a0-b7dc-42b7601967f0">

Please see the following reference for more details on how it is / was used in Microsoft:
[Agile Project Management with Kanban used in some Teams in Microsoft](https://www.amazon.com/Project-Management-Kanban-Developer-Practices/dp/0735698953). 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by erinmez** [comment]
Hi @rileybroughten, do you have any thoughts on this?

**Reply by queenofcorgis** [reply]
Hi there,

Thanks for leaving this feedback and sharing such detailed thoughts. While the GitHub product team does evaluate the feedback we receive from the community, we are not able to respond to every submission.

Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our Code of Conduct for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

**Reply by erinmez** [reply]
Hi there,

I was not aware that this would be interpreted as unsolicited because the product manager was involved in the earlier discussions [here](https://github.com/orgs/community/discussions/4848).

I apologize for any trouble this may have caused.

Best regards,
Tanju

**Reply by zsloane** [reply]
Are there any guidelines as to what is considered an "unsolicited mention"? 

I have mentioned the most recent communicator on a topic in the past. Particularly when there has been a long period of silence. 

I want to make sure I am not violating the Code of Conduct. Is it best to just avoid @mentions in general here?

**Reply by queenofcorgis** [reply]
@zsloane thanks for asking 💟 ! Generally, we'd recommend avoiding pinging or @mentioning someone (both staff or member) who is not currently part of the conversation on that specific post. So if you @ mentioned me in your comment on this Discussion that would be within our guidelines. The reasons being if everyone started pinging folks from our most helpful and our staff nobody would get anything else done and we simply cannot respond to every post. 

And to set expectations, [we have some guidelines here](https://github.com/community/community?tab=readme-ov-file#from-a-suggestion-to-a-shipped-feature) about what happens after you share a request or bug. We know this information isn't in the most easily findable spot and we're actively  working on updating on how/where this information is shared so members know what to expect when posting.

**Reply by zsloane** [reply]
Understood. Thank you for clarifying!
